src/App.tsx

```diff
@@ -47,9 +51,152 @@ function App() {
 
     }, [selectedFile]);
 
+
+    function isLastDay(date: string) {
+        let d = new Date(date)
+        let test = new Date(d.getTime())
+        let month = test.getMonth();
+        test.setDate(test.getDate() + 1);
+        return test.getMonth() !== month;
+    }
```

Comment:
This method is a nice helper function to get the last day of the month. 
The calculation is also a valuable approach to get the las day of the month. 

I'd use this method more generic, to get the last day of the selected timespan. 
So you can save some lines of code and tests further down.

I'd go with something like this: 

```
const isLastDayOf = (dateString: string, dateResolution: IDateResolution) => {
        if (dateResolution === 'daily') {
            return true;
        }

        if (dateResolution === 'weekly') {
            return dayjs(dateString).format('d') === '0';
        }

        return ...
};
```

and also there is a nice approach dayjs offers you to compare against the last day of month: 

```

dayjs(dateString).format('DD') === dayjs(dateString).endOf('month').format('DD');

```